R/quote.R

```diff
@@ -121,17 +130,87 @@ quote_identifier <-
 
 #' @rdname hidden_aliases
 #' @export
-setMethod("dbQuoteIdentifier", c("DBIConnection"), quote_identifier)
+setMethod("dbQuoteIdentifier", signature("DBIConnection"), quote_identifier)
 
 # Need to keep other method declarations around for now, because clients might
 # use getMethod(), see e.g. https://github.com/r-dbi/odbc/pull/149
 #' @rdname hidden_aliases
 #' @export
-setMethod("dbQuoteIdentifier", c("DBIConnection", "character"), quote_identifier)
+setMethod("dbQuoteIdentifier", signature("DBIConnection", "character"), quote_identifier)
 
 #' @rdname hidden_aliases
 #' @export
-setMethod("dbQuoteIdentifier", c("DBIConnection", "SQL"), quote_identifier)
+setMethod("dbQuoteIdentifier", signature("DBIConnection", "SQL"), quote_identifier)
+
+#' Unquote identifiers
+#'
+#' Call this method to convert a [SQL] object created by [dbQuoteIdentifier()]
+#' back to a list of [Table] objects.
+#'
+#' @param conn A subclass of [DBIConnection-class], representing
+#'   an active connection to an DBMS.
+#' @param x An [SQL] or [Table] object or character vector, or a list of such
+#'   objects, to unquote.
+#' @param ... Other arguments passed on to methods.
+#'
+#' @template methods
+#' @templateVar method_name dbUnquoteIdentifier
+#'
+#' @inherit DBItest::spec_sql_unquote_identifier return
+#' @inheritSection DBItest::spec_sql_unquote_identifier Specification
+#'
+#' @family DBIResult generics
+#' @export
+#' @examples
+#' # Unquoting allows to understand the structure of a possibly complex quoted
+#' # identifier
+#'
+#' dbUnquoteIdentifier(
+#'   ANSI(),
+#'   SQL(c("Schema"."Table", "UnqualifiedTable"))
```

Comment:
Unless I am missing something this does not parse. I think you are missing a set of quotes.

```r
SQL(c("Schema"."Table", "UnqualifiedTable"))
#> Error: unexpected symbol in "SQL(c("Schema"."
```